### ISSUE TITLE
App not opened on answer after unlocking ios device

### ISSUE THREAD

**Issue by tuneerclixlogix** [react-native-webrtc/react-native-callkeep#355]
When the device is locked and there is any incoming call, i answered the call and then unlocked the screen but the application was in background state

Expected Behaviour: After unlocking the device it should be in the foreground state .

**Comment by umairlive490** [comment]
Same issue for me.
When the device is locked and there is any incoming call, i answered the call and then unlocked the screen but the application was in background state.After unlocking the device it should be in the foreground state .

**Comment by tuneerclixlogix** [comment]
Did you found any resolution ? @umairlive490

**Comment by m4r00p** [comment]
Yes, any luck?

**Comment by timorss** [comment]
Is there a solution?

**Comment by gabrielelana** [comment]
Solved this yesterday, got an hint from the WhatsApp behaviour

TLDR

```javascript
  RNCallKeep.displayIncomingCall(
    UUID, 
    'Caller Number', 
    'Caller Name',
    'generic',
    true  // <- THIS IS THE MAGIC
  );
```

Apparently you need to tell iOS that your call is a video call, and since (I guess) it has no way to tell how do you want to display the video streams, it will bring the specific application in foreground.

The following is the behaviour that I managed to have (close to what I have on Android): 

> Ring -> Answer -> Unlock -> Application

Before I was only able to have:

> Ring -> Answer -> Native Call UI -> Press Application Button (bottom right) -> Unlock -> Application

A huge improvement in usability

**Comment by KeKam** [comment]
> Solved this yesterday, got an hint from the WhatsApp behaviour
> 
> TLDR
> 
> ```js
>   RNCallKeep.displayIncomingCall(
>     UUID, 
>     'Caller Number', 
>     'Caller Name',
>     'generic',
>     true  // <- THIS IS THE MAGIC
>   );
> ```
> 
> Apparently you need to tell iOS that your call is a video call, and since (I guess) it has no way to tell how do you want to display the video streams, it will bring the specific application in foreground.
> 
> The following is the behaviour that I managed to have (close to what I have on Android):
> 
> > Ring -> Answer -> Unlock -> Application
> 
> Before I was only able to have:
> 
> > Ring -> Answer -> Native Call UI -> Press Application Button (bottom right) -> Unlock -> Application
> 
> A huge improvement in usability

Would setting `hadVideo: YES` in `reportNewIncomingCall` within `AppDelegate.m` not achieve the same behaviour? Cause it doesn't.

**Comment by gabrielelana** [comment]
> Wouldn't setting `hadVideo: YES` in `reportNewIncomingCall` within `AppDelegate.m` not achieve the same behaviour? Cause it doesn't.

You mean `hasVideo: YES`? 

Don't know, didn't tried that, I use data-only push notifications, not voip notifications

**Comment by fukemy** [comment]
thanks @gabrielelana , it's really importaint answer to me. love u so much <3

**Comment by fukemy** [comment]
@gabrielelana , not worked to me. I changed hasVideo = true in AppDelegate, but still not working, I am using pushkit

**Comment by gabrielelana** [comment]
@fukemy I don't know exactly what you do in AppDelegate, but note that what works for me is probably different: I set the flag when I need to display the incoming call `RNCallKeep.displayIncomingCall` and I do that when I receive the push notification. Hope this helps

**Comment by MrGru** [comment]
If you are using `reportNewIncomingCall` in AppDelegate. you should change: hasVideo = true, fromPushKit: NO

```
[RNCallKeep reportNewIncomingCall:uuid
                               handle:@"app name"
                           handleType:@"generic"
                             hasVideo:true <---- change this line
                  localizedCallerName:@"caller name"
                      supportsHolding:NO
                         supportsDTMF:NO
                     supportsGrouping:NO
                   supportsUngrouping:NO
                          fromPushKit:NO <---- change this line
                              payload:NULL
                withCompletionHandler:completion
```

and use  imageName = icon user can click to open the app, in the setup function: [the icon should be transparent](https://stackoverflow.com/questions/45581940/app-icon-is-not-showing-in-callkit-ui)
```
[RNCallKeep setup:@{
     @"appName": @"Video Visit",
     @"imageName": @"call.png", <--- Chagne this line
     @"maximumCallGroups": @1,
     @"maximumCallsPerCallGroup": @1,
     @"supportsVideo": @YES, <--- Chagne this line
     @"displayCallReachabilityTimeout":@60000
   }];
```

**Comment by AndreasKarlzzon** [comment]
+1, any news for doese that has not got it to open the app after unlocking the screen on received call?

**Comment by aromero-braintly** [comment]
> Solved this yesterday, got an hint from the WhatsApp behaviour
> 
> TLDR
> 
> ```js
>   RNCallKeep.displayIncomingCall(
>     UUID, 
>     'Caller Number', 
>     'Caller Name',
>     'generic',
>     true  // <- THIS IS THE MAGIC
>   );
> ```
> 
> Apparently you need to tell iOS that your call is a video call, and since (I guess) it has no way to tell how do you want to display the video streams, it will bring the specific application in foreground.
> 
> The following is the behaviour that I managed to have (close to what I have on Android):
> 
> > Ring -> Answer -> Unlock -> Application
> 
> Before I was only able to have:
> 
> > Ring -> Answer -> Native Call UI -> Press Application Button (bottom right) -> Unlock -> Application
> 
> A huge improvement in usability

Hi, this worked for me on version 4.1.0 but won't work on 4.3.1. I needed to update to be able to route audio through bluetooth, so it's not an option for me to downgrade.
Did anyone have better luck trying to solve this?

**Comment by MinhThu100200** [comment]
@AndreasKarlzzon I think because you try to turn off the default call screen by call endCall event when answering the calling.  I faced this issue so I still keep default call screen
